src/DOM/__tests__/hydration.spec.tsx

```diff
@@ -122,16 +116,18 @@ describe('SSR Hydration - (JSX)', () => {
 			expect2: '<div><em>Works <span>again</span>!</em></div>'
 		}
 	].forEach(({ node, expect1, expect2 }, i) => {
-		it(`Validate various structures #${ (i + 1) }`, () => {
+		it(`Validate various structures #${ (i + 1) }`, (done) => {
 			const html = renderToString(node);
 			const container = createContainerWithHTML(html);
 
 			expect(innerHTML(container.innerHTML)).to.equal(innerHTML(expect1));
 			render(node, container);
-			expect(validateNodeTree(node)).to.equal(true);
-			expect(innerHTML(container.innerHTML)).to.equal(innerHTML(expect2));
-			render(node, container);
-			expect(innerHTML(container.innerHTML)).to.equal(innerHTML(expect2));
+
+			setTimeout (function () {
```

Comment:
We could attach a ref to the last node in the tree that will signal when hydration is complete, then we can use requestAnimationFrame without breaking the tests.